File: VerbumKit/Sources/Clients/Fixtures/EntityFixtureData.swift
import Models

// FIXTURE — hand-curated. Labelled as such; nothing here is production content.
//
// The first entity coverage the spec suggests (§68) plus the few people, events
// and passages the golden path needs (§75: David → Goliath → 1 Samuel 17).
// Summaries are short and non-interpretive (§3.5); dates are hedged. Every
// relationship carries a source (§33). Real content arrives with Task 11.
enum EntityFixtureData {
    // MARK: Sources

    static let scriptureSource = SourceReference(
        id: "fixture.source.web",
        citation: "World English Bible (public domain) — the passages cited",
        url: "https://worldenglish.bible"
    )
    static let editorialSource = SourceReference(
        id: "fixture.source.editorial",
        citation: "Verbum editorial notes (fixture; to be replaced by sourced content)",
        url: nil
    )
    static let sources: [SourceReference] = [scriptureSource, editorialSource]

    // MARK: Entities

    private static func person(_ key: String, _ name: String, _ summary: String) -> BibleEntity { .init(id: "fixture.person.\(key)", type: .person, name: name, summary: summary) }
    private static func place(_ key: String, _ name: String, _ summary: String) -> BibleEntity { .init(id: "fixture.place.\(key)", type: .place, name: name, summary: summary) }
    private static func theme(_ key: String, _ name: String, _ summary: String) -> BibleEntity { .init(id: "fixture.theme.\(key)", type: .theme, name: name, summary: summary) }
    private static func event(_ key: String, _ name: String, _ summary: String) -> BibleEntity { .init(id: "fixture.event.\(key)", type: .event, name: name, summary: summary) }

    /// Passages appear in the graph as nodes so "key passages" are edges like any other.
    static func passageNode(_ reference: PassageReference) -> BibleEntity {
        BibleEntity(id: "passage.\(reference.bookId).\(reference.chapter)", type: .passage, name: reference.formatted(for: .english), summary: nil)
    }

    static func passageReference(for node: BibleEntity) -> PassageReference? {
        guard node.type == .passage else { return nil }
        let parts = node.id.split(separator: ".")
        guard parts.count == 3, parts[0] == "passage", let chapter = Int(parts[2]) else { return nil }
        return PassageReference(bookId: String(parts[1]), chapter: chapter)
    }

    static let entities: [BibleEntity] = [
        // People (§68) + golden-path people
        person("jesus", "Jesus", "Central figure of the New Testament."),
        person("paul", "Paul", "Apostle; author of many New Testament letters."),
        person("peter", "Peter", "Disciple of Jesus; leader among the apostles."),
        person("abraham", "Abraham", "Patriarch; the covenant with God begins with him."),
        person("moses", "Moses", "Led Israel out of Egypt; received the Law."),
        person("david", "David", "King of Israel; associated with many Psalms."),
        person("solomon", "Solomon", "King of Israel; built the first temple."),
        person("saul", "Saul", "First king of Israel."),
        person("mary", "Mary", "Mother of Jesus."),
        person("john", "John", "Disciple of Jesus; traditionally linked to the Gospel of John."),
        person("goliath", "Goliath", "Philistine champion from Gath, defeated by David."),
        person("samuel", "Samuel", "Prophet and judge; anointed Saul and David."),
        person("bathsheba", "Bathsheba", "Wife of Uriah, then of David; mother of Solomon."),
        // Places (§68)
        place("jerusalem", "Jerusalem", "City of David; site of the temple."),
        place("bethlehem", "Bethlehem", "Birthplace of David and of Jesus."),
        place("nazareth", "Nazareth", "Town in Galilee where Jesus grew up."),
        place("galilee", "Galilee", "Northern region; setting of much of Jesus' ministry."),
        place("rome", "Rome", "Capital of the empire; destination of Paul's letter to the Romans."),
        place("corinth", "Corinth", "Greek city; recipient of two of Paul's letters."),
        place("ephesus", "Ephesus", "City in Asia Minor; recipient of Paul's letter to the Ephesians."),
        place("babylon", "Babylon", "Empire and city; place of Judah's exile."),
        place("egypt", "Egypt", "Where Israel was enslaved before the Exodus."),
        place("elah", "Valley of Elah", "Where Israel and the Philistines faced each other in 1 Samuel 17."),
        // Themes (§68)
        theme("faith", "Faith", "Trust in God; a recurring theme across both testaments."),
        theme("grace", "Grace", "Unearned favour; central in Paul's letters."),
        theme("forgiveness", "Forgiveness", "Release from wrongdoing, divine and human."),
        theme("love", "Love", "Of God, and for neighbour; the greatest commandments."),
        theme("anxiety", "Anxiety", "Worry and its answer; e.g. Matthew 6:25–34."),
        theme("wisdom", "Wisdom", "Skill in living well; Proverbs, Job, Ecclesiastes."),
        theme("justice", "Justice", "Right dealing, especially toward the vulnerable."),
        theme("prayer", "Prayer", "Speaking with God; Psalms, the Lord's Prayer."),
        theme("money", "Money", "Wealth, generosity and its dangers."),
        theme("suffering", "Suffering", "Pain and its meaning; Job, Lamentations, the Passion."),
        // Events
        event("david-anointed", "Anointing of David", "Samuel anoints David in Bethlehem while Saul is still king."),
        event("david-goliath", "David and Goliath", "David defeats the Philistine champion in the Valley of Elah."),
        event("david-takes-jerusalem", "David takes Jerusalem", "David captures the Jebusite stronghold and makes it his capital."),
        event("exodus", "The Exodus", "Israel leaves Egypt under Moses."),
        // Passage nodes
        passageNode(.init(bookId: "Gen", chapter: 1)),
        passageNode(.init(bookId: "1Sam", chapter: 16)),
        passageNode(.init(bookId: "1Sam", chapter: 17)),
        passageNode(.init(bookId: "2Sam", chapter: 5)),
        passageNode(.init(bookId: "Ps", chapter: 23)),
        passageNode(.init(bookId: "Ps", chapter: 51)),
        passageNode(.init(bookId: "Matt", chapter: 6)),
        passageNode(.init(bookId: "John", chapter: 3)),
        passageNode(.init(bookId: "Rom", chapter: 8)),
    ]

    private static let byID: [EntityID: BibleEntity] = Dictionary(uniqueKeysWithValues: entities.map { ($0.id, $0) })
    static func entity(_ id: EntityID) -> BibleEntity? { byID[id] }

    // MARK: Relationships (§22.2), each with a source (§33)

    private static func edge(_ source: String, _ type: RelationshipType, _ target: String, scripture: Bool = true) -> BibleRelationship {
        BibleRelationship(
            id: "fixture.edge.\(source).\(type.rawValue).\(target)",
            sourceId: source, targetId: target, type: type,
            confidence: scripture ? 1.0 : 0.8,
            sourceReferenceIds: [scripture ? scriptureSource.id : editorialSource.id]
        )
    }
    private static let P = "fixture.person.", L = "fixture.place.", T = "fixture.theme.", E = "fixture.event."

    static let relationships: [BibleRelationship] = [
        // David cluster — the golden path
        edge(P + "david", .appearsIn, "passage.1Sam.16"),
        edge(P + "david", .appearsIn, "passage.1Sam.17"),
        edge(P + "david", .appearsIn, "passage.2Sam.5"),
        edge(P + "david", .appearsIn, "passage.Ps.23"),
        edge(P + "david", .appearsIn, "passage.Ps.51"),
        edge(P + "david", .relatedTo, P + "goliath"),
        edge(P + "david", .relatedTo, P + "saul"),
        edge(P + "david", .relatedTo, P + "samuel"),
        edge(P + "david", .relatedTo, P + "bathsheba"),
        edge(P + "david", .relatedTo, P + "solomon"),
        edge(P + "david", .participatesIn, E + "david-anointed"),
        edge(P + "david", .participatesIn, E + "david-goliath"),
        edge(P + "david", .participatesIn, E + "david-takes-jerusalem"),
        edge(P + "david", .relatedTo, L + "bethlehem"),
        edge(P + "david", .relatedTo, L + "jerusalem"),
        edge(P + "david", .relatedToTheme, T + "forgiveness", scripture: false),
        edge(P + "david", .relatedToTheme, T + "prayer", scripture: false),
        edge(P + "goliath", .appearsIn, "passage.1Sam.17"),
        edge(P + "goliath", .participatesIn, E + "david-goliath"),
        edge(P + "saul", .appearsIn, "passage.1Sam.16"),
        edge(P + "saul", .appearsIn, "passage.1Sam.17"),
        edge(P + "saul", .relatedTo, P + "samuel"),
        edge(P + "samuel", .appearsIn, "passage.1Sam.16"),
        edge(P + "samuel", .participatesIn, E + "david-anointed"),
        edge(P + "bathsheba", .relatedTo, P + "solomon"),
        edge(P + "solomon", .relatedTo, L + "jerusalem"),
        edge(E + "david-anointed", .occursAt, L + "bethlehem"),
        edge(E + "david-anointed", .precedes, E + "david-goliath"),
        edge(E + "david-goliath", .occursAt, L + "elah"),
        edge(E + "david-goliath", .precedes, E + "david-takes-jerusalem"),
        edge(E + "david-goliath", .relatedToTheme, T + "faith", scripture: false),
        edge(E + "david-takes-jerusalem", .occursAt, L + "jerusalem"),
        edge(E + "david-takes-jerusalem", .appearsIn, "passage.2Sam.5"),
        edge("passage.Ps.51", .relatedToTheme, T + "forgiveness", scripture: false),
        edge("passage.Ps.23", .relatedToTheme, T + "prayer", scripture: false),
        // Jesus cluster
        edge(P + "jesus", .appearsIn, "passage.John.3"),
        edge(P + "jesus", .appearsIn, "passage.Matt.6"),
        edge(P + "jesus", .relatedTo, P + "mary"),
        edge(P + "jesus", .relatedTo, P + "peter"),
        edge(P + "jesus", .relatedTo, P + "john"),
        edge(P + "jesus", .relatedTo, L + "bethlehem"),
        edge(P + "jesus", .relatedTo, L + "nazareth"),
        edge(P + "jesus", .relatedTo, L + "galilee"),
        edge(P + "jesus", .relatedTo, L + "jerusalem"),
        edge("passage.Matt.6", .relatedToTheme, T + "anxiety", scripture: false),
        edge("passage.Matt.6", .relatedToTheme, T + "money", scripture: false),
        edge("passage.Matt.6", .relatedToTheme, T + "prayer", scripture: false),
        edge("passage.John.3", .relatedToTheme, T + "love", scripture: false),
        edge("passage.John.3", .relatedToTheme, T + "faith", scripture: false),
        // Paul cluster
        edge(P + "paul", .appearsIn, "passage.Rom.8"),
        edge(P + "paul", .relatedTo, L + "rome"),
        edge(P + "paul", .relatedTo, L + "corinth"),
        edge(P + "paul", .relatedTo, L + "ephesus"),
        edge("passage.Rom.8", .relatedToTheme, T + "grace", scripture: false),
        edge("passage.Rom.8", .relatedToTheme, T + "suffering", scripture: false),
        // Moses / Abraham
        edge(P + "moses", .participatesIn, E + "exodus"),
        edge(P + "moses", .relatedTo, L + "egypt"),
        edge(E + "exodus", .occursAt, L + "egypt"),
        edge(P + "abraham", .relatedToTheme, T + "faith", scripture: false),
    ]

    // MARK: Details (§9)

    private static func detail(_ id: String, aliases: [String] = [], dates: String? = nil, role: String? = nil, modern: String? = nil, passages: [PassageReference] = []) -> EntityDetail {
        EntityDetail(entity: byID[id]!, aliases: aliases, approximateDates: dates, role: role, modernGeography: modern, keyPassages: passages, sources: sources)
    }

    static let details: [EntityID: EntityDetail] = Dictionary(uniqueKeysWithValues: [
        detail(P + "david", dates: "c. 1010–970 BC (commonly dated; the chronology is approximate)", role: "Second king of Israel",
               passages: [.init(bookId: "1Sam", chapter: 16), .init(bookId: "1Sam", chapter: 17), .init(bookId: "2Sam", chapter: 5), .init(bookId: "Ps", chapter: 23), .init(bookId: "Ps", chapter: 51)]),
        detail(P + "goliath", role: "Philistine champion from Gath", passages: [.init(bookId: "1Sam", chapter: 17)]),
        detail(P + "saul", dates: "late 11th century BC (commonly dated)", role: "First king of Israel", passages: [.init(bookId: "1Sam", chapter: 16), .init(bookId: "1Sam", chapter: 17)]),
        detail(P + "samuel", role: "Prophet and last of the judges", passages: [.init(bookId: "1Sam", chapter: 16)]),
        detail(P + "solomon", dates: "c. 970–931 BC (commonly dated)", role: "Third king of Israel"),
        detail(P + "jesus", aliases: ["Jesus of Nazareth", "Christ", "Messiah"], dates: "c. 4 BC – c. AD 30/33 (scholars vary)", role: "Central figure of the New Testament",
               passages: [.init(bookId: "John", chapter: 3), .init(bookId: "Matt", chapter: 6)]),
        detail(P + "paul", aliases: ["Saul of Tarsus"], dates: "c. AD 5 – c. 64/67 (scholars vary)", role: "Apostle to the Gentiles", passages: [.init(bookId: "Rom", chapter: 8)]),
        detail(P + "peter", aliases: ["Simon", "Cephas"], role: "Disciple and apostle"),
        detail(P + "moses", role: "Prophet; leader of the Exodus"),
        detail(P + "abraham", aliases: ["Abram"], role: "Patriarch"),
        detail(L + "jerusalem", aliases: ["Jebus", "Salem", "Zion"], modern: "Jerusalem, Israel/Palestine", passages: [.init(bookId: "2Sam", chapter: 5)]),
        detail(L + "bethlehem", aliases: ["Ephrath"], modern: "Bethlehem, West Bank", passages: [.init(bookId: "1Sam", chapter: 16)]),
        detail(L + "elah", modern: "Wadi es-Sunt, Israel (commonly identified)", passages: [.init(bookId: "1Sam", chapter: 17)]),
        detail(L + "egypt", modern: "Egypt"),
        detail(L + "rome", modern: "Rome, Italy", passages: [.init(bookId: "Rom", chapter: 8)]),
        detail(E + "david-goliath", dates: "during Saul's reign (commonly dated to the late 11th century BC)", passages: [.init(bookId: "1Sam", chapter: 17)]),
        detail(E + "david-anointed", dates: "during Saul's reign", passages: [.init(bookId: "1Sam", chapter: 16)]),
        detail(E + "david-takes-jerusalem", dates: "early in David's reign (commonly dated c. 1000 BC)", passages: [.init(bookId: "2Sam", chapter: 5)]),
        detail(E + "exodus", dates: "disputed; commonly placed in the 15th or 13th century BC"),
        detail(T + "forgiveness", passages: [.init(bookId: "Ps", chapter: 51)]),
        detail(T + "anxiety", passages: [.init(bookId: "Matt", chapter: 6, verses: 25...34)]),
        detail(T + "prayer", passages: [.init(bookId: "Ps", chapter: 23), .init(bookId: "Matt", chapter: 6, verses: 5...15)]),
        detail(T + "grace", passages: [.init(bookId: "Rom", chapter: 8)]),
        detail(T + "faith", passages: [.init(bookId: "1Sam", chapter: 17), .init(bookId: "John", chapter: 3)]),
        detail(T + "love", passages: [.init(bookId: "John", chapter: 3, verses: 16...16)]),
    ].map { ($0.entity.id, $0) })
}

File: android/core/clients/src/main/kotlin/com/nexussoft/verbum/clients/fixtures/EntityFixtureData.kt
package com.nexussoft.verbum.clients.fixtures

import com.nexussoft.verbum.models.BibleEntity
import com.nexussoft.verbum.models.BibleEntityType
import com.nexussoft.verbum.models.BibleRelationship
import com.nexussoft.verbum.models.BookLanguage
import com.nexussoft.verbum.models.EntityDetail
import com.nexussoft.verbum.models.EntityId
import com.nexussoft.verbum.models.PassageReference
import com.nexussoft.verbum.models.RelationshipType
import com.nexussoft.verbum.models.SourceReference
import com.nexussoft.verbum.models.formatted

// FIXTURE — hand-curated, ported from the iOS EntityFixtureData (same data, one origin).
// Labelled as such; nothing here is production content. Summaries are short and
// non-interpretive (§3.5); dates are hedged; every relationship carries a source (§33).
internal object EntityFixtureData {
    val scriptureSource = SourceReference("fixture.source.web", "World English Bible (public domain) — the passages cited", "https://worldenglish.bible")
    val editorialSource = SourceReference("fixture.source.editorial", "Verbum editorial notes (fixture; to be replaced by sourced content)", null)
    val sources = listOf(scriptureSource, editorialSource)

    fun passageNode(reference: PassageReference) =
        BibleEntity("passage.${reference.bookId}.${reference.chapter}", BibleEntityType.PASSAGE, reference.formatted(BookLanguage.ENGLISH), null)

    fun passageReference(node: BibleEntity): PassageReference? {
        if (node.type != BibleEntityType.PASSAGE) return null
        val parts = node.id.split(".")
        if (parts.size != 3 || parts[0] != "passage") return null
        return PassageReference(parts[1], parts[2].toIntOrNull() ?: return null)
    }

    val entities: List<BibleEntity> = listOf(
        BibleEntity("fixture.person.jesus", BibleEntityType.PERSON, "Jesus", "Central figure of the New Testament."),
        BibleEntity("fixture.person.paul", BibleEntityType.PERSON, "Paul", "Apostle; author of many New Testament letters."),
        BibleEntity("fixture.person.peter", BibleEntityType.PERSON, "Peter", "Disciple of Jesus; leader among the apostles."),
        BibleEntity("fixture.person.abraham", BibleEntityType.PERSON, "Abraham", "Patriarch; the covenant with God begins with him."),
        BibleEntity("fixture.person.moses", BibleEntityType.PERSON, "Moses", "Led Israel out of Egypt; received the Law."),
        BibleEntity("fixture.person.david", BibleEntityType.PERSON, "David", "King of Israel; associated with many Psalms."),
        BibleEntity("fixture.person.solomon", BibleEntityType.PERSON, "Solomon", "King of Israel; built the first temple."),
        BibleEntity("fixture.person.saul", BibleEntityType.PERSON, "Saul", "First king of Israel."),
        BibleEntity("fixture.person.mary", BibleEntityType.PERSON, "Mary", "Mother of Jesus."),
        BibleEntity("fixture.person.john", BibleEntityType.PERSON, "John", "Disciple of Jesus; traditionally linked to the Gospel of John."),
        BibleEntity("fixture.person.goliath", BibleEntityType.PERSON, "Goliath", "Philistine champion from Gath, defeated by David."),
        BibleEntity("fixture.person.samuel", BibleEntityType.PERSON, "Samuel", "Prophet and judge; anointed Saul and David."),
        BibleEntity("fixture.person.bathsheba", BibleEntityType.PERSON, "Bathsheba", "Wife of Uriah, then of David; mother of Solomon."),
        BibleEntity("fixture.place.jerusalem", BibleEntityType.PLACE, "Jerusalem", "City of David; site of the temple."),
        BibleEntity("fixture.place.bethlehem", BibleEntityType.PLACE, "Bethlehem", "Birthplace of David and of Jesus."),
        BibleEntity("fixture.place.nazareth", BibleEntityType.PLACE, "Nazareth", "Town in Galilee where Jesus grew up."),
        BibleEntity("fixture.place.galilee", BibleEntityType.PLACE, "Galilee", "Northern region; setting of much of Jesus' ministry."),
        BibleEntity("fixture.place.rome", BibleEntityType.PLACE, "Rome", "Capital of the empire; destination of Paul's letter to the Romans."),
        BibleEntity("fixture.place.corinth", BibleEntityType.PLACE, "Corinth", "Greek city; recipient of two of Paul's letters."),
        BibleEntity("fixture.place.ephesus", BibleEntityType.PLACE, "Ephesus", "City in Asia Minor; recipient of Paul's letter to the Ephesians."),
        BibleEntity("fixture.place.babylon", BibleEntityType.PLACE, "Babylon", "Empire and city; place of Judah's exile."),
        BibleEntity("fixture.place.egypt", BibleEntityType.PLACE, "Egypt", "Where Israel was enslaved before the Exodus."),
        BibleEntity("fixture.place.elah", BibleEntityType.PLACE, "Valley of Elah", "Where Israel and the Philistines faced each other in 1 Samuel 17."),
        BibleEntity("fixture.theme.faith", BibleEntityType.THEME, "Faith", "Trust in God; a recurring theme across both testaments."),
        BibleEntity("fixture.theme.grace", BibleEntityType.THEME, "Grace", "Unearned favour; central in Paul's letters."),
        BibleEntity("fixture.theme.forgiveness", BibleEntityType.THEME, "Forgiveness", "Release from wrongdoing, divine and human."),
        BibleEntity("fixture.theme.love", BibleEntityType.THEME, "Love", "Of God, and for neighbour; the greatest commandments."),
        BibleEntity("fixture.theme.anxiety", BibleEntityType.THEME, "Anxiety", "Worry and its answer; e.g. Matthew 6:25–34."),
        BibleEntity("fixture.theme.wisdom", BibleEntityType.THEME, "Wisdom", "Skill in living well; Proverbs, Job, Ecclesiastes."),
        BibleEntity("fixture.theme.justice", BibleEntityType.THEME, "Justice", "Right dealing, especially toward the vulnerable."),
        BibleEntity("fixture.theme.prayer", BibleEntityType.THEME, "Prayer", "Speaking with God; Psalms, the Lord's Prayer."),
        BibleEntity("fixture.theme.money", BibleEntityType.THEME, "Money", "Wealth, generosity and its dangers."),
        BibleEntity("fixture.theme.suffering", BibleEntityType.THEME, "Suffering", "Pain and its meaning; Job, Lamentations, the Passion."),
        BibleEntity("fixture.event.david-anointed", BibleEntityType.EVENT, "Anointing of David", "Samuel anoints David in Bethlehem while Saul is still king."),
        BibleEntity("fixture.event.david-goliath", BibleEntityType.EVENT, "David and Goliath", "David defeats the Philistine champion in the Valley of Elah."),
        BibleEntity("fixture.event.david-takes-jerusalem", BibleEntityType.EVENT, "David takes Jerusalem", "David captures the Jebusite stronghold and makes it his capital."),
        BibleEntity("fixture.event.exodus", BibleEntityType.EVENT, "The Exodus", "Israel leaves Egypt under Moses."),
        passageNode(PassageReference("Gen", 1)),
        passageNode(PassageReference("1Sam", 16)),
        passageNode(PassageReference("1Sam", 17)),
        passageNode(PassageReference("2Sam", 5)),
        passageNode(PassageReference("Ps", 23)),
        passageNode(PassageReference("Ps", 51)),
        passageNode(PassageReference("Matt", 6)),
        passageNode(PassageReference("John", 3)),
        passageNode(PassageReference("Rom", 8)),
    )

    private val byId: Map<EntityId, BibleEntity> = entities.associateBy { it.id }
    fun entity(id: EntityId): BibleEntity? = byId[id]

    private fun edge(source: String, type: RelationshipType, target: String, scripture: Boolean = true) = BibleRelationship(
        id = "fixture.edge.$source.${type.wireValue}.$target",
        sourceId = source, targetId = target, type = type,
        confidence = if (scripture) 1.0 else 0.8,
        sourceReferenceIds = listOf(if (scripture) scriptureSource.id else editorialSource.id),
    )

    val relationships: List<BibleRelationship> = listOf(
        edge("fixture.person.david", RelationshipType.APPEARS_IN, "passage.1Sam.16"),
        edge("fixture.person.david", RelationshipType.APPEARS_IN, "passage.1Sam.17"),
        edge("fixture.person.david", RelationshipType.APPEARS_IN, "passage.2Sam.5"),
        edge("fixture.person.david", RelationshipType.APPEARS_IN, "passage.Ps.23"),
        edge("fixture.person.david", RelationshipType.APPEARS_IN, "passage.Ps.51"),
        edge("fixture.person.david", RelationshipType.RELATED_TO, "fixture.person.goliath"),
        edge("fixture.person.david", RelationshipType.RELATED_TO, "fixture.person.saul"),
        edge("fixture.person.david", RelationshipType.RELATED_TO, "fixture.person.samuel"),
        edge("fixture.person.david", RelationshipType.RELATED_TO, "fixture.person.bathsheba"),
        edge("fixture.person.david", RelationshipType.RELATED_TO, "fixture.person.solomon"),
        edge("fixture.person.david", RelationshipType.PARTICIPATES_IN, "fixture.event.david-anointed"),
        edge("fixture.person.david", RelationshipType.PARTICIPATES_IN, "fixture.event.david-goliath"),
        edge("fixture.person.david", RelationshipType.PARTICIPATES_IN, "fixture.event.david-takes-jerusalem"),
        edge("fixture.person.david", RelationshipType.RELATED_TO, "fixture.place.bethlehem"),
        edge("fixture.person.david", RelationshipType.RELATED_TO, "fixture.place.jerusalem"),
        edge("fixture.person.david", RelationshipType.RELATED_TO_THEME, "fixture.theme.forgiveness", scripture = false),
        edge("fixture.person.david", RelationshipType.RELATED_TO_THEME, "fixture.theme.prayer", scripture = false),
        edge("fixture.person.goliath", RelationshipType.APPEARS_IN, "passage.1Sam.17"),
        edge("fixture.person.goliath", RelationshipType.PARTICIPATES_IN, "fixture.event.david-goliath"),
        edge("fixture.person.saul", RelationshipType.APPEARS_IN, "passage.1Sam.16"),
        edge("fixture.person.saul", RelationshipType.APPEARS_IN, "passage.1Sam.17"),
        edge("fixture.person.saul", RelationshipType.RELATED_TO, "fixture.person.samuel"),
        edge("fixture.person.samuel", RelationshipType.APPEARS_IN, "passage.1Sam.16"),
        edge("fixture.person.samuel", RelationshipType.PARTICIPATES_IN, "fixture.event.david-anointed"),
        edge("fixture.person.bathsheba", RelationshipType.RELATED_TO, "fixture.person.solomon"),
        edge("fixture.person.solomon", RelationshipType.RELATED_TO, "fixture.place.jerusalem"),
        edge("fixture.event.david-anointed", RelationshipType.OCCURS_AT, "fixture.place.bethlehem"),
        edge("fixture.event.david-anointed", RelationshipType.PRECEDES, "fixture.event.david-goliath"),
        edge("fixture.event.david-goliath", RelationshipType.OCCURS_AT, "fixture.place.elah"),
        edge("fixture.event.david-goliath", RelationshipType.PRECEDES, "fixture.event.david-takes-jerusalem"),
        edge("fixture.event.david-goliath", RelationshipType.RELATED_TO_THEME, "fixture.theme.faith", scripture = false),
        edge("fixture.event.david-takes-jerusalem", RelationshipType.OCCURS_AT, "fixture.place.jerusalem"),
        edge("fixture.event.david-takes-jerusalem", RelationshipType.APPEARS_IN, "passage.2Sam.5"),
        edge("passage.Ps.51", RelationshipType.RELATED_TO_THEME, "fixture.theme.forgiveness", scripture = false),
        edge("passage.Ps.23", RelationshipType.RELATED_TO_THEME, "fixture.theme.prayer", scripture = false),
        edge("fixture.person.jesus", RelationshipType.APPEARS_IN, "passage.John.3"),
        edge("fixture.person.jesus", RelationshipType.APPEARS_IN, "passage.Matt.6"),
        edge("fixture.person.jesus", RelationshipType.RELATED_TO, "fixture.person.mary"),
        edge("fixture.person.jesus", RelationshipType.RELATED_TO, "fixture.person.peter"),
        edge("fixture.person.jesus", RelationshipType.RELATED_TO, "fixture.person.john"),
        edge("fixture.person.jesus", RelationshipType.RELATED_TO, "fixture.place.bethlehem"),
        edge("fixture.person.jesus", RelationshipType.RELATED_TO, "fixture.place.nazareth"),
        edge("fixture.person.jesus", RelationshipType.RELATED_TO, "fixture.place.galilee"),
        edge("fixture.person.jesus", RelationshipType.RELATED_TO, "fixture.place.jerusalem"),
        edge("passage.Matt.6", RelationshipType.RELATED_TO_THEME, "fixture.theme.anxiety", scripture = false),
        edge("passage.Matt.6", RelationshipType.RELATED_TO_THEME, "fixture.theme.money", scripture = false),
        edge("passage.Matt.6", RelationshipType.RELATED_TO_THEME, "fixture.theme.prayer", scripture = false),
        edge("passage.John.3", RelationshipType.RELATED_TO_THEME, "fixture.theme.love", scripture = false),
        edge("passage.John.3", RelationshipType.RELATED_TO_THEME, "fixture.theme.faith", scripture = false),
        edge("fixture.person.paul", RelationshipType.APPEARS_IN, "passage.Rom.8"),
        edge("fixture.person.paul", RelationshipType.RELATED_TO, "fixture.place.rome"),
        edge("fixture.person.paul", RelationshipType.RELATED_TO, "fixture.place.corinth"),
        edge("fixture.person.paul", RelationshipType.RELATED_TO, "fixture.place.ephesus"),
        edge("passage.Rom.8", RelationshipType.RELATED_TO_THEME, "fixture.theme.grace", scripture = false),
        edge("passage.Rom.8", RelationshipType.RELATED_TO_THEME, "fixture.theme.suffering", scripture = false),
        edge("fixture.person.moses", RelationshipType.PARTICIPATES_IN, "fixture.event.exodus"),
        edge("fixture.person.moses", RelationshipType.RELATED_TO, "fixture.place.egypt"),
        edge("fixture.event.exodus", RelationshipType.OCCURS_AT, "fixture.place.egypt"),
        edge("fixture.person.abraham", RelationshipType.RELATED_TO_THEME, "fixture.theme.faith", scripture = false),
    )

    private fun detail(id: String, aliases: List<String> = emptyList(), dates: String? = null, role: String? = null, modern: String? = null, passages: List<PassageReference> = emptyList()) =
        EntityDetail(byId.getValue(id), aliases, dates, role, modern, passages, sources)

    val details: Map<EntityId, EntityDetail> = listOf(
        detail("fixture.person.david", dates = "c. 1010–970 BC (commonly dated; the chronology is approximate)", role = "Second king of Israel", passages = listOf(PassageReference("1Sam", 16), PassageReference("1Sam", 17), PassageReference("2Sam", 5), PassageReference("Ps", 23), PassageReference("Ps", 51))),
        detail("fixture.person.goliath", role = "Philistine champion from Gath", passages = listOf(PassageReference("1Sam", 17))),
        detail("fixture.person.saul", dates = "late 11th century BC (commonly dated)", role = "First king of Israel", passages = listOf(PassageReference("1Sam", 16), PassageReference("1Sam", 17))),
        detail("fixture.person.samuel", role = "Prophet and last of the judges", passages = listOf(PassageReference("1Sam", 16))),
        detail("fixture.person.solomon", dates = "c. 970–931 BC (commonly dated)", role = "Third king of Israel"),
        detail("fixture.person.jesus", aliases = listOf("Jesus of Nazareth", "Christ", "Messiah"), dates = "c. 4 BC – c. AD 30/33 (scholars vary)", role = "Central figure of the New Testament", passages = listOf(PassageReference("John", 3), PassageReference("Matt", 6))),
        detail("fixture.person.paul", aliases = listOf("Saul of Tarsus"), dates = "c. AD 5 – c. 64/67 (scholars vary)", role = "Apostle to the Gentiles", passages = listOf(PassageReference("Rom", 8))),
        detail("fixture.person.peter", aliases = listOf("Simon", "Cephas"), role = "Disciple and apostle"),
        detail("fixture.person.moses", role = "Prophet; leader of the Exodus"),
        detail("fixture.person.abraham", aliases = listOf("Abram"), role = "Patriarch"),
        detail("fixture.place.jerusalem", aliases = listOf("Jebus", "Salem", "Zion"), modern = "Jerusalem, Israel/Palestine", passages = listOf(PassageReference("2Sam", 5))),
        detail("fixture.place.bethlehem", aliases = listOf("Ephrath"), modern = "Bethlehem, West Bank", passages = listOf(PassageReference("1Sam", 16))),
        detail("fixture.place.elah", modern = "Wadi es-Sunt, Israel (commonly identified)", passages = listOf(PassageReference("1Sam", 17))),
        detail("fixture.place.egypt", modern = "Egypt"),
        detail("fixture.place.rome", modern = "Rome, Italy", passages = listOf(PassageReference("Rom", 8))),
        detail("fixture.event.david-goliath", dates = "during Saul's reign (commonly dated to the late 11th century BC)", passages = listOf(PassageReference("1Sam", 17))),
        detail("fixture.event.david-anointed", dates = "during Saul's reign", passages = listOf(PassageReference("1Sam", 16))),
        detail("fixture.event.david-takes-jerusalem", dates = "early in David's reign (commonly dated c. 1000 BC)", passages = listOf(PassageReference("2Sam", 5))),
        detail("fixture.event.exodus", dates = "disputed; commonly placed in the 15th or 13th century BC"),
        detail("fixture.theme.forgiveness", passages = listOf(PassageReference("Ps", 51))),
        detail("fixture.theme.anxiety", passages = listOf(PassageReference("Matt", 6, 25..34))),
        detail("fixture.theme.prayer", passages = listOf(PassageReference("Ps", 23), PassageReference("Matt", 6, 5..15))),
        detail("fixture.theme.grace", passages = listOf(PassageReference("Rom", 8))),
        detail("fixture.theme.faith", passages = listOf(PassageReference("1Sam", 17), PassageReference("John", 3))),
        detail("fixture.theme.love", passages = listOf(PassageReference("John", 3, 16..16))),
    ).associateBy { it.entity.id }
}
